kyo-core/shared/src/test/scala/kyoTest/envsTest.scala

```diff
@@ -40,7 +40,18 @@ class envsTest extends KyoTest:
         succeed
     }
 
-    "invalid inference" in pendingUntilFixed {
+    "reduce large intersection" in {
+        val t1: Int < Envs[Int & String & Boolean & Float & Char & Double] = 18
+        val t2                                                             = Envs[Int].run(42)(t1)
+        val t3 = Envs[String].run("a")(t2)
+        val t4 = Envs[Boolean].run(false)(t3)
+        val t5 = Envs[Float].run(0.23f)(t4)
+        val t6 = Envs[Char].run('a')(t5)
+        val t7 = Envs[Double].run(0.23d)(t6)
```

Comment:
If you structure the runs like:

```
Envs[Char & Double].run('a') {
  Envs[Double].run(0.23d)(???)
}
```
Does the type inference work correctly? I think it would be good to have a test for this.